src/Plugin/Block/BaristaBlock.php

```diff
@@ -0,0 +1,162 @@
+   * @var \Drupal\draco_coffee\DracoCoffeeManager
+   */
+  protected $dracoCoffeeManager;
+
+  /**
+   * The route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The State API service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * Constructs a new ExampleBlock instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration, i.e. an array with configuration values keyed
+   *   by configuration option name. The special key 'context' may be used to
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The cofiguration factory service.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The current user service.
+   * @param \Drupal\draco_coffee\DracoCoffeeManager $draco_coffee_manager
+   *   The Draco Coffee Manager service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The State API service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory, AccountProxyInterface $current_user, DracoCoffeeManager $draco_coffee_manager, EntityTypeManagerInterface $entity_type_manager, RouteMatchInterface $route_match, StateInterface $state) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+    $this->currentUser = $current_user;
+    $this->dracoCoffeeManager = $draco_coffee_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->routeMatch = $route_match;
+    $this->state = $state;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('current_user'),
+      $container->get('draco_coffee.manager'),
+      $container->get('entity_type.manager'),
+      $container->get('current_route_match'),
+      $container->get('state')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function blockAccess(AccountInterface $account) {
+    // Display the block only for users with a particular role.
+    if ($account->hasPermission('administer draco_coffee configuration')) {
+      return AccessResult::allowed()
+        ->addCacheContexts(['route.name'])
```

Comment:
I'm not getting why we're varying the cache by route? 😕

Reply:
Also, shouldn't this cache metadata be part of the render array output in the `build` method? 

Have a look at `AccessResult::allowedIfHasPermission` to simplify.

Reply:
True, I can remove the route cacheability. Notice that I still don't grok Cache API so sometimes I don't know what I am doing. Thanks for the help! I will look into that method.

Reply:
I fixed this by using a custom tag. The reason is that cacheability depends on a couple state variables. Does this make sense to you or is there a better way to do it @m4olivei?

Reply:
That makes sense to me, yes.